sdk/cosmos/src/clients/cosmos_client.rs

```diff
@@ -41,17 +41,17 @@ pub struct CosmosOptions {
 }
 
 impl CosmosOptions {
-    /// TODO
-    pub fn with_client(client: Arc<Box<dyn HttpClient>>) -> Self {
-        Self {
-            retry: RetryOptions::new(3),
-            transport: TransportOptions::new(move |_ctx, req| {
-                let client = client.clone();
-                let req = req.take_inner();
-                Box::pin(async move { Ok(client.execute_request(req).await?.into()) })
-            }),
-        }
-    }
+    // TODO
+    // pub fn with_client(client: Arc<Box<dyn HttpClient>>) -> Self {
```

Comment:
CosmosOption::with_client from #204 Policy Architecture does not compile now and I could use some help, @rylev.

Reply:
Can you post the compiler error?

Reply:
```
PS C:\Users\cataggar\io\azure-sdk-for-rust> cargo check
    Checking azure_cosmos v0.1.0 (C:\Users\cataggar\io\azure-sdk-for-rust\sdk\cosmos)
error: future cannot be sent between threads safely
  --> sdk\cosmos\src\clients\cosmos_client.rs:51:17
   |
51 |                 Box::pin(async move { Ok(client.execute_request(req).await?.into()) })
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ future created by async block is not `Send`
   |
   = help: the trait `std::marker::Send` is not implemented for `dyn futures::Future<Output = Result<http::Response<bytes::Bytes>, Box<dyn StdError + Sync + std::marker::Send>>>`
note: future is not `Send` as it awaits another future which is not `Send`
  --> sdk\cosmos\src\clients\cosmos_client.rs:51:42
   |
51 |                 Box::pin(async move { Ok(client.execute_request(req).await?.into()) })
   |                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^ await occurs here on type `Pin<Box<dyn futures::Future<Output = Result<http::Response<bytes::Bytes>, Box<dyn StdError + Sync + std::marker::Send>>>>>`, which is not `Send`
   = note: required for the cast to the object type `dyn futures::Future<Output = Result<azure_core::Response, Box<(dyn StdError + Sync + std::marker::Send + 'static)>>> + std::marker::Send`
```